[ML] Single Metric Viewer: Fix time bounds with custom strings.

walterra opened the pull request:
## Summary

Fixes #54903.
Part of #52986.

Makes sure to set `bounds` via `timefilter.getBounds()` again and not infer directly from `globalState` to correctly consider custom strings like `now-15m`.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20836/)
* Commit: 8263dd3f8d4c70ebec630b54b8145cdb89580d69

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20836/","number":"20836","commit":"8263dd3f8d4c70ebec630b54b8145cdb89580d69"}]}
PIPELINE-->

walterra commented:
Thanks @jgowdyelastic @peteharverson for the review! I added another commit 5f4d4c1 after I found out that time strings which are also invalid for `timefilter` (e.g. something like `now-asdf15min`) would also break the page. This is now fixed by not passing on an `undefined` for `min/max` of bounds to the inner components. I also added a unit test for `TimeSeriesExplorerUrlStateManager`. The test isn't able to cover the particular issue that this PR is solving but at least it's a first test to run the component properly. The code also adds a few checks against `timefilter` methods when they could return `undefined`.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20904/)
* Commit: 5f4d4c13d6a40693a5325d8afbc21c91946e56a4

### History
* :green_heart: [Build #20836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20836/) succeeded 8263dd3f8d4c70ebec630b54b8145cdb89580d69

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20836/","number":"20836","commit":"8263dd3f8d4c70ebec630b54b8145cdb89580d69"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20904/","number":"20904","commit":"5f4d4c13d6a40693a5325d8afbc21c91946e56a4"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/21100/)
* Commit: 5f4d4c13d6a40693a5325d8afbc21c91946e56a4

### History
* :green_heart: [Build #20904](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20904/) succeeded 5f4d4c13d6a40693a5325d8afbc21c91946e56a4
* :green_heart: [Build #20836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20836/) succeeded 8263dd3f8d4c70ebec630b54b8145cdb89580d69

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20836/","number":"20836","commit":"8263dd3f8d4c70ebec630b54b8145cdb89580d69"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/20904/","number":"20904","commit":"5f4d4c13d6a40693a5325d8afbc21c91946e56a4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/21100/","number":"21100","commit":"5f4d4c13d6a40693a5325d8afbc21c91946e56a4"}]}
PIPELINE-->
